### PR TITLE
removed faulty and unnecessary values mechanism

### PR DESCRIPTION
**Pull request by Morriz**
Hi, I removed the faulty and unnecessary values mechanism, allowing for empty value triggering.

I also removed the Changeset convenience methods which were not needed since the Changeset now only consists of the id => values map.

Hope you agree with the changes. Please don't hesitate to ask any questions.

### CODE REVIEW

**Comment by Morriz** [comment]
This also closes #11 :)

**Comment by mrfelton** [comment]
@Morriz I'd prefer that we limit this patch to just resolving the bug in question. Could you separate out the bug fix into a simpler patch please that doesn't also include the more comprehensive changes that you have made here.

If you would like to discuss changing the Changeset API could you create a separate issue and patch?That type of change could potentially break things for existing users.

**Comment by Morriz** [comment]
I do however think that the changes I made were necessary, as the code was breaking under normal circumstances. That rendered at least the value convenience methods invalid. They had no right of existence as is imo. Imo the message format should only be a hash with id => instance (why not use the full blown instance since we have it already, it makes no sense to have to retrieve them later on if needed for property relationship checking).

I really have no idea why you made such decisions imo, but I think the software will get better when multiple people contribute. You can put back the convenience methods (wth a deprecated warning) to accomodate ppl depending on it, but I doubt they are out there.

Anyways, I will build out the version I think it should be, and use it myself, as I have a direct need for it. Thanks for the initial version :)

**Comment by beeman** [comment]
This PR won't be accepted as it removes functionality that our codebase relies on, without an alternative. Additionally @mrfelton and I have already been thinking about and working on changing the behaviour and structure of the application in a upcoming version  https://github.com/fullcube/loopback-ds-changed-mixin/commit/b7912f22343adb1deb113a821cc7aa9e211bcb6d .

The functionality that this PR removes is the following. Consider the output below. There is a fair use case for being able to get a list of all changed items, based on the value they changed to. 

In this example we want some special action to happen if a status changes to `pending`. If you only have the `id` : `value` pair as in this PR, you will need to loop through all the items, and filter the ones with the status you want. It makes total sense to have this mixin give you a complete image of the changeset, so you are flexible in what you do with these changes.

```
{
  status: {
    ids: {
      '5586c58848fb091e068f8115': 'pending',
      '5586c58848fb091e068f8116': 'pending'
    },
    values: {
      pending: [
        '5586c58848fb091e068f8115',
        '5586c58848fb091e068f8116'
      ]
    }
  }
}
```

Long story short, a PR that fixes the issue with null values will be absolutely considered, but not a refactor that removes functionality.

**Comment by Morriz** [comment]
I don't agree with you argument since the functionality it depends on is faulty. Plus, a seasoned programmer does not need such sugar, and can do that with simple funcional programming.
But, like I said, your assumption is based on a faulty assumption that values can't be empty...
